packages/asc-ui/src/components/Checkbox/Checkbox.tsx

```diff
@@ -19,13 +25,23 @@ const Checkbox = React.forwardRef<
       variant,
       disabled,
       error,
+      indeterminate,
       ...otherProps
     },
-    ref,
+    externalRef,
   ) => {
     const [checked, setChecked] = useState(!!checkedProp)
     const [focus, setFocus] = useState(false)
     const { setActive } = useContext(LabelContext)
+    const ref = React.useRef<HTMLInputElement>(null)
+
+    useImperativeHandle(externalRef, () => ref.current as HTMLInputElement)
+
+    useEffect(() => {
+      if (ref.current && typeof indeterminate === 'boolean') {
+        ref.current.indeterminate = indeterminate
+      }
```

Comment:
If the `indeterminate` prop was `true` and then becomes `undefined` it will stay `true` since the type is checked. Best to default to `false` here if `indeterminate` is not set.

```suggestion
      if (ref.current) {
        ref.current.indeterminate = indeterminate ?? false
      }
```

Reply:
Nice one, I should start to use this operator more :)